.github/workflows/validate.yml

```diff
@@ -16,50 +20,52 @@ on:
 
 jobs:
   validate:
-    name: Validate ubuntu-latest ghc-${{ matrix.ghc }}
+    name: Validate ${{ matrix.os }} ghc-${{ matrix.ghc }}
 
-    runs-on: ubuntu-latest
-    continue-on-error: ${{ matrix.experimental }}
+    runs-on: ${{ matrix.os }}
+    continue-on-error: ${{ matrix.experimental == 'true' }}
 
     strategy:
+      # To remove after test workflow
+      fail-fast: false
       matrix:
+        os: ["ubuntu-latest", "macos-latest", "windows-latest"]
+        ghc: ["9.2.1", "9.0.2", "9.0.1", "8.10.7", "8.8.4", "8.6.5", "8.4.4", "8.0.2"]
         include:
-          - ghc: "9.2.1"
-            cli: true
-            experimental: true # work to be done
-            flags: ""
-          - ghc: "9.0.2"
-            cli: false
-            experimental: true # see #7987
-            flags: "--lib-only"
-          - ghc: "9.0.1"
-            cli: false
-            experimental: false
-            flags: "--lib-only"
-          - ghc: "8.10.7"
-            cli: true
-            experimental: false
-            flags: ""
-          - ghc: "8.8.4"
-            cli: true
-            experimental: false
-            flags: "--solver-benchmarks"
-          - ghc: "8.6.5"
-            cli: true
-            experimental: false
-            flags: "--complete-hackage-tests"
-          - ghc: "8.4.4"
-            cli: true
-            experimental: false
-            flags: ""
-          - ghc: "8.2.2"
-            cli: true
-            experimental: false
-            flags: ""
-          - ghc: "8.0.2"
-            cli: false
-            experimental: false
-            flags: "--lib-only"
+          - os: "ubuntu-latest"
+            ghc: "9.2.1"
+            experimental: "true" # work to be done
+          - os: "macos-latest"
+            ghc: "9.2.1"
+            experimental: "true" # work to be done
+          - os: "ubuntu-latest"
+            ghc: "9.0.2"
+            cli: "false"
+            experimental: "true" # see #7987
+          - os: "macos-latest"
+            ghc: "9.0.2"
+            cli: "false"
+            experimental: "true" # see #7987
+          - os: "macos-latest"
+            ghc: "9.0.1"
+            cli: "false"
+            experimental: "true" # see #7988
+          - os: "ubuntu-latest"
+            ghc: "9.0.1"
+            cli: "false"
+          - os: "macos-latest"
+            ghc: "8.10.7"
+            experimental: "true" # see #7952 and #7988
+          - os: "ubuntu-latest"
+            ghc: "8.0.2"
+            cli: "false"
+          - os: "macos-latest"
+            ghc: "8.0.2"
+            cli: "false"
+            experimental: "true" # see #7988 and #7989
+          - os: "windows-latest"
+            ghc: "8.0.2"
+            cli: "false"
```

Comment:
I don't understand how you are building this matrix. Do you want to list cases one by one or do a full cross-product, or a combination of both? Eg. ubuntu-latest, ghc 8.0.2 is included twice, once with `cli: false` and the other with `cli` undefined (AFAIU)

Reply:
agaik if you repeat params from the product it only adds the new param, you csn check in the workflow run there sre no repeared jobs